tests/e2e/__testers__.ts

```diff
@@ -35,32 +74,26 @@ export const testTemplate = (params: {
             },
           },
         }) as ClientBase
-      const getPrismaAdmin = async () =>
-        new (await getPrismaClientModule()).PrismaClient({
-          datasources: {
-            db: {
-              url: `${databaseUrlBase}/postgres`,
-            },
-          },
-        }) as ClientBase
+
+      const getPrismaAdmin = async () => {
+        return params.getPrismaAdmin(databaseUrlBase, (await getPrismaClientModule()).PrismaClient)
+      }
 
       const dropTestDatabase = async () => {
-        const prisma = await getPrismaAdmin()
-        try {
-          await prisma.$executeRawUnsafe(`DROP DATABASE ${databaseName} WITH (FORCE);`)
-        } catch (error) {
-          const isDatabaseNotFoundErorr = error instanceof Error && error.message.match(/does not exist/)
-          if (!isDatabaseNotFoundErorr) throw error
-        } finally {
-          await prisma.$disconnect()
-        }
+        await ctx.run(`prisma migrate reset --force`, { reject: true })
+        return dropDatabase(await getPrismaAdmin(), databaseName, datasourceProvider)
```

Comment:
```suggestion
        return await dropDatabase(await getPrismaAdmin(), databaseName, datasourceProvider)
```

Changes behaviour of `throw` in async code.

Reply:
From my understanding there is no difference in those two. `return promise` and `return await promise` do the same thing here.

What am I missing?

Reply:
IIRC there are subtle differences, but don't have time to fetch links to those past issues. Feel free to ignore this one.

Reply:
The difference is that if you have a finally as in try/catch/finally, then that will be executed _before_ the Promise finishes, if you don't have the await.

Reply:
@endor that is correct. At the moment there is not catch or finally here and ideally we want to catch it outside and react on it.

Example for that specific case (which honestly doesn't look like a good looking code):
```
async function rejectionWithReturnAwait () {
  try {
    return await Promise.reject(new Error())
  } catch (e) {
    return 'Saved!'
  }
}

async function rejectionWithReturn () {
  try {
    return Promise.reject(new Error())
  } catch (e) {
    return 'Catch is not triggered!'
  }
}
```